test/test_dbcore.py

```diff
@@ -760,8 +762,31 @@ def test_no_results(self):
             ModelFixture1, dbcore.query.FalseQuery()).get())
 
 
+class ParentalDirCreation(unittest.TestCase):
+    @mock.patch('builtins.input', side_effect=['y', ])
+    def test_create_yes(self, _):
+        non_exist_path = "ParentalDirCreationTest/nonexist/" + str(random())
+        try:
+            dbcore.Database(non_exist_path)
+        except OSError as e:
+            raise e
+        shutil.rmtree("ParentalDirCreationTest")
+
+    @mock.patch('builtins.input', side_effect=['n', ])
```

Comment:
It might be easier to use the `control_stdin` context manager from `test/helper.py` (see its docstring for details).